nengo/networks/assoc_mem.py

```diff
@@ -75,9 +75,12 @@ def __init__(self, input_vectors, output_vectors=None,  # noqa: C901
         if is_iterable(output_vectors):
             output_vectors = np.matrix(output_vectors)
 
-        # Fail if number of input items and number of output items don't
-        # match
-        if input_vectors.shape[0] != output_vectors.shape[0]:
+        if input_vectors.shape[0] <= 0:
+            # Fail if number of input items <= 0
+            raise ValueError('Number of input vectors cannot be <= 0.')
```

Comment:
Also, maybe put a note like "If using a Vocabulary, make sure it is not empty". In fact, I might prefer to have a separate check for Vocabularies, so that we can give a tailored error message.